### ISSUE TITLE
Could not find any cudnn.h matching version '8' in any subdirectory

### ISSUE THREAD

**Issue by Thunder003** [tensorflow/tensorflow#44574]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):
- TensorFlow version: 1.14
- Python version: 3.6
- Installed using virtualenv? pip? conda?: 
- Bazel version (if compiling from source): 0.24.1
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 11.1
- GPU model and memory: RTX 3090

**Describe the problem**

For some specific purpose, I am building TF 1.14 from source with CUDA 11.1, CUDNN 8.0.4 & TensorRT 7.2 on Ubuntu 16.04 but getting an error:-
![image](https://user-images.githubusercontent.com/51915348/98066809-20352b00-1e7e-11eb-8365-906eeb4dfa5b.png)

I have made various checks like LD_LIBRARY_PATH, PATH are properly set and installation using tar, deb file of CUDNN but the error persists. Even if CUDA 11.1, CUDNN 8.0.4 is incompatible with TF 1.14, why I'm getting this error as this error shows unable to find the file. Maybe incompatibility error will generate during compilation.

**Provide the exact sequence of commands / steps that you executed before running into the problem**

**Any other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by amahendrakar** [comment]
@Thunder003,
TensorFlow 1.x is not actively supported. Every TensorFlow release is compatible with a certain CUDA/cuDNN version. Please take a look at the [tested build configurations](https://www.tensorflow.org/install/source#gpu) for more information. 

Also, please check similar issue [#42895](https://github.com/tensorflow/tensorflow/issues/42895#issuecomment-685874736) for reference. Thanks!

**Comment by Thunder003** [comment]
@amahendrakar let me rephrase my issue. When building Tensorflow 2 from source with CUDA 11.1, CUDNN 8.0.4, what TensorFlow 2.x checks at this stage( Configuration step) **inside the CUDNN file** which helps it recognize the CUDNN version. I have verified that there is no such issue with T.F. 2.3, CUDA 11.1, and CUDNN 8.0.4 
I know the original issue is due to incompatible, but can you please give some insights by relating it to the CUDNN file( like what is checks inside it).  As T.F. 1.14 had recognized the CUDA 11.1 file.

**Comment by rmothukuru** [comment]
@Thunder003,
Checks are done in this [Build file](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/stream_executor/cuda/BUILD). 

To elaborate, the `Version` of `CUDNN` is identified from [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/stream_executor/cuda/BUILD#L404). After that, compatibility is checked in [this line](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/stream_executor/cuda/BUILD#L637), which in turn calls `cudnn_version.cc` where [compatibility is checked](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/stream_executor/cuda/cudnn_version.cc#L21-L32).

Hope this helps. Thanks!

**Comment by Thunder003** [comment]
@rmothukuru thanks for your answer.

**Comment by smitheric95** [comment]
Hey @Thunder003, were you ever able to get TF 1.14 to build with CUDA 11.1? I'm currently trying to do the same with TF 1.15.

**Comment by Thunder003** [comment]
@smitheric95 no (I tried almost everything, but T.F. 1.14 & CUDA 11.1 are incompatible),  every time I try to build T.F. 1.14 & CUDA 11.1, I get an error:- ""Could not find any cudnn.h matching version '8' in any subdirectory""

I eventually migrated from T.F. 1.x to 2.x.

**Comment by smitheric95** [comment]
@Thunder003, ah yes. I encountered that as well, but was able to get around it by changing `configure.py` to look for the CUDNN version in `cudnn_version.h` instead of `cudnn.h`.

**Comment by Thunder003** [comment]
@smitheric95 ok, I have contacted a few people, saying to me that T.F. 1.14 & CUDA 11.1 are incompatible so may get an error on compilation time even I had fixed this issue. But another thing that was scratching my head is that if they are incompatible, then why NGC provide a container having T.F. 1.15 and CUDA 11.1. 
Try compiling that @smitheric95, if you will be able to make that, let me know here as well :)

**Comment by smitheric95** [comment]
@Thunder003, that worked! 

I just had to copy the `tensorflow` directory from the docker container to my local `/opt/` directory. 

Then, I set `IN_CONTAINER` to `0` on line 108 of `nvbuild.sh`.

Finally, I ran:
```bash
export CUDA_VERSION="11.1"
export CUDNN_VERSION="8.0.5"
./nvbuild.sh --v1
```

You may run into some permissions issues along the way – I handled those as they came up.

**Comment by Thunder003** [comment]
@smitheric95 nice!!, thanks for sharing as well,..... which docker container you are referring to, NGC T.F. 1.15, CUDA 11.1? And have you tried running it, is it utilizing GPUs properly?

**Comment by smitheric95** [comment]
@Thunder003, correct it's the `20.11-tf1-py3` container on NGC. I used that source code and built 1.15.4 natively. 

I can confirm that the GPUs are being properly utilized as well.

**Comment by NoaBrazilay** [comment]
Hi @smitheric95,
Thanks for your help. Can you elaborate on where I can find the "TensorFlow directory from the docker"?
Should I get it from the container or from the source-code? Can you provide a link to this source code?

Thanks.

**Comment by Thunder003** [comment]
hi @smitheric95, just checking, have you tested any code with your build from source TensorFlow? Is it working fine?

**Comment by smitheric95** [comment]
Hey @Thunder003, yes, I have tested TensorFlow code with this setup and am able to successfully train models just fine! 

As far as StyleGAN goes, you might want to test some other simpler models first to rule out environment issues.

**Comment by ShiningDeadFishEyes** [comment]
> Hey @Thunder003, yes, I have tested TensorFlow code with this setup and am able to successfully train models just fine!
> 
> As far as StyleGAN goes, you might want to test some other simpler models first to rule out environment issues.

Hi,sorry to bother you ,but I was trying to compile tensorflow 1.15 with 3090 and cuda11.0 ,however the script report that it could not found libcurand.so.11. I tried to find "libcurand" with find /usr/local -type f | grep libcurand , but I found libcurand.so.10 instead of 11 in cuda11 folder. Have you encountered this problem？ Or could you plz list what did you do to   built 1.15.4 successfully? Thanks!

**Comment by ssrokyz** [comment]
In the case of TF 2.0, I should have changed cudnn.h -> cudnn_version.h in the "third_party/gpus/find_cuda_config.py" file

**Comment by bio-punk** [comment]
As same as TF2.2,thx

> In the case of TF 2.0, I should have changed cudnn.h -> cudnn_version.h in the "third_party/gpus/find_cuda_config.py" file
